internal/domain/review_suggestion.go

```diff
@@ -0,0 +1,27 @@
+package domain
+
+import (
+	"github.com/invopop/jsonschema"
+)
+
+type ReviewSuggestion struct {
+	Comment  string `json:"comment"`
+	Position int    `json:"position"`
+}
+
+type ReviewSuggestionsResponse struct {
+	ReviewSuggestions []ReviewSuggestion `json:"review_suggestions"`
+}
+
+func GenerateSchema[T any]() interface{} {
```

Comment:
The `jsonschema` library usage assumes reflection will effectively work for all types. Verify if there's a need for additional configurations or consider adding error handling for potential reflection errors, which can avoid runtime panics if unexpected struct types are passed.